src/LogFileReader.cpp

```diff
@@ -74,12 +81,18 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
-
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
-    return true;
+    TEMP_TIME = line.section(",",0,0);
+    QTime bdt = QTime::fromString(TEMP_TIME, TIME_FORMAT);
+    double bdv= line.section(", ",1,1).toDouble();
+    double bdc = line.section(", ",2,2).toDouble();
+    if (bdt.isNull() || !bdv || !bdc){
+        qDebug() << "\nError reading from data"<<endl;
```

Comment:
This debug while a good idea, is not really necessary since the this function is not in charge of telling the user whether there was a problem or not, it just has to tell the calling function whether there was a problem or not. So just return false is enough :)